#! /bin/sh /usr/share/dpatch/dpatch-run
## 14size_matters.dpatch by  <hesso@pool.math.tu-berlin.de>
##
## DP: * Increase the MAXSTR macro to suit present-day needs.
## DP: * Increase the maximum length of the input line buffer likewise.

@DPATCH@
diff -Naur screen-4.0.3.orig/input.c screen-4.0.3/input.c
--- screen-4.0.3.orig/input.c	2008-06-01 23:39:49.000000000 +0200
+++ screen-4.0.3/input.c	2008-06-01 23:40:37.000000000 +0200
@@ -38,7 +38,7 @@
 
 struct inpline
 {
-  char  buf[101];	/* text buffer */
+  char  buf[MAXSTR+1];	/* text buffer */
   int  len;		/* length of the editible string */
   int  pos;		/* cursor position in editable string */
 };
@@ -49,7 +49,7 @@
 struct inpdata
 {
   struct inpline inp;
-  int  inpmaxlen;	/* 100, or less, if caller has shorter buffer */
+  int  inpmaxlen;	/* MAXSTR, or less, if caller has shorter buffer */
   char *inpstring;	/* the prompt */
   int  inpstringlen;	/* length of the prompt */
   int  inpmode;		/* INP_NOECHO, INP_RAW, INP_EVERY */
@@ -120,8 +120,8 @@
   int maxlen;
   struct inpdata *inpdata;
   
-  if (len > 100)
-    len = 100;
+  if (len > MAXSTR)
+    len = MAXSTR;
   if (!(mode & INP_NOECHO))
     {
       maxlen = flayer->l_width - 1 - strlen(istr);
diff -Naur screen-4.0.3.orig/process.c screen-4.0.3/process.c
--- screen-4.0.3.orig/process.c	2008-06-01 23:39:49.000000000 +0200
+++ screen-4.0.3/process.c	2008-06-01 23:40:37.000000000 +0200
@@ -1934,7 +1934,7 @@
 	ChangeAKA(fore, *args, strlen(*args));
       break;
     case RC_COLON:
-      Input(":", 100, INP_COOKED, Colonfin, NULL, 0);
+      Input(":", MAXSTR, INP_COOKED, Colonfin, NULL, 0);
       if (*args && **args)
 	{
 	  s = *args;
diff -Naur screen-4.0.3.orig/screen.h screen-4.0.3/screen.h
--- screen-4.0.3.orig/screen.h	2008-06-01 23:39:49.000000000 +0200
+++ screen-4.0.3/screen.h	2008-06-01 23:40:37.000000000 +0200
@@ -86,7 +86,7 @@
 
 #define Ctrl(c) ((c)&037)
 
-#define MAXSTR		256
+#define MAXSTR		768
 #define MAXARGS 	64
 #define MSGWAIT 	5
 #define MSGMINWAIT 	1
diff -Naur screen-4.0.3.orig/window.c screen-4.0.3/window.c
--- screen-4.0.3.orig/window.c	2008-06-01 23:39:49.000000000 +0200
+++ screen-4.0.3/window.c	2008-06-01 23:40:37.000000000 +0200
@@ -2171,7 +2171,7 @@
       return;
     }
   flayer = &p->w_layer;
-  Input(":", 100, INP_COOKED, zmodem_fin, NULL, 0);
+  Input(":", MAXSTR, INP_COOKED, zmodem_fin, NULL, 0);
   s = send ? zmodem_sendcmd : zmodem_recvcmd;
   n = strlen(s);
   LayProcess(&s, &n);
